docs/proposals/scalability-testing.md

```diff
@@ -0,0 +1,78 @@
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/scalability-testing.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Background
+
+We have a goal to be able to scale to 1000-node clusters by end of 2015.
+However, running regression tests on 1000-node clusters or requiring developers
+to test their changes (with respect to performance) on 1000-node clusters is
+ridiculous. So we need some infrastructure to simulate big clusters on relatively
+small number of machines.
+
+
+## Proposal
+
+In this proposal we are focusing on scalability testing of the whole cluster.
+We do NOT focus on node-scalability - this issue should be handled separately.
+
+Since we do not focus on the node performance, we don't need real Kubelet and
+KubeProxy - in fact we don't even need to start real containers.
+All we actually need is to have some Kubelet-like and KubeProxy-like components
+that will be simulating the load on apiserver that their real equivalents are
+generating (e.g. sending NodeStatus updated, watching for pods, watching for
+endpoints (KubeProxy), etc.).
+
+What needs to be done:
+1. Determine what requests both KubeProxy and Kubelet are sending to apiserver.
+2. Create a KubeletSim that is generating the same load on apiserver that the
+   real Kubelet, but is not starting any containers. In the initial version we
+   can assume that pods never die, so it is enough to just react on the state
+   changes read from apiserver.
+	 TBD: Maybe we can reuse a real Kubelet for it by just injecting some "fake"
+   interfaces to it?
+3. Similarly create a KubeProxySim that is generating the same load on apiserver
+   that the real one. Again, since we are not planning to talk to those
+   containers, it basically doesn't need to do anything apart from that.
+	 TBD: Maybe we can reuse a real KubeProxy for it by just injecting some "fake"
+   interfaces to it?
+4. Refactor kube-up/kube-down scripts (or create new ones) to allow starting
+   a cluster with KubeletSim and KubeProxySim instead of real ones and put
+   a bunch of them on a single machine.
+5. Create a load generator for it (probably initially it would be enough to
```

Comment:
I'm not sure if I understood your comment.
The load generator I'm talking above is not for Kubelet - it will just be generating load on apiserver (e.g. creating/updating/deleting pods or RCs). Those pods will then be pick by KubeletSims. I would like to avoid creating anything very sophisticated in the first pass to just prove that this approach is useful.